apps/dcos_l4lb/src/dcos_l4lb_mesos_poller.erl

```diff
@@ -125,9 +125,15 @@ is_healthy(_Task) ->
 collect_port_mappings(Tasks) ->
     maps:fold(fun (_TaskId, Task, Acc) ->
         [TaskIP | _TaskIPs] = maps:get(task_ip, Task),
-        PMs = [{{Protocol, Host}, {TaskIP, Port}}
-              || #{host_port := Host, protocol := Protocol,
-                   port := Port} <- maps:get(ports, Task, [])],
+        AgentIP = maps:get(agent_ip, Task),
+        PMs = [{{Protocol, Host},
+                case maps:get(container, Task) of
+                    <<"DOCKER">> ->
+                         {AgentIP, Host};
```

Comment:
It's not a port mapping. To fix the issue, we should use port mappings only for mesos runtime, i.e.

```
Runtime = maps:get(runtime, Task),
[ {{Protocol, Host}, {TaskIP, Port}}
|| #{host_port := Host, protocol := Protocol,
     port := Port} <- maps:get(ports, Task, []),
   Runtime =/= docker ],
```

Reply:
Good suggestion!